courses/serializers.py

```diff
@@ -39,15 +40,52 @@ def to_internal_value(self, value):
         return obj
 
 
+class StringyManyToManyField(serializers.Field):
+    """
+    Serializes out as a list of strings on the way out, but reifies into
+    real models on the way in.
+    """
+    def __init__(self, *args, **kwargs):
+        self.model = kwargs.pop('model', None)
+        self.prop = kwargs.pop('lookup', None)
+
+        if not self.model:
+            raise ImproperlyConfigured("You must provide a model to use for reifying")
+
+        if not self.prop:
+            raise ImproperlyConfigured("You must provide a field lookup to "
+                                       "convert string value into model instance.")
+
+        super(StringyManyToManyField, self).__init__(*args, **kwargs)
+
+    def to_representation(self, obj):
+        """Python object to string"""
+        return (str(i) for i in obj.all())
+
+    def to_internal_value(self, value):
+        """Incoming value to python value"""
+        if not isinstance(value, list):
+            value = [value]
```

Comment:
This also will fail if the input is a generator or something else that's not a list